File: med/python/Question1343.py
from typing import List


class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        ans, aver, k_threshold = 0, 0, k * threshold
        for i in range(len(arr)):
            aver += arr[i]
            if i < k - 1:
                continue
            else:
                if aver >= k_threshold:
                    ans += 1
                aver -= arr[i - k + 1]
        return ans


print(Solution().numOfSubarrays([2, 2, 2, 2, 5, 5, 5, 8], 3, 4))

File: cpp/med/Question1343.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int numOfSubarrays(vector<int> &arr, int k, int threshold) {
        int ans = 0, k_thre = k * threshold, sum = 0;
        for (int i = 0; i < arr.size(); i++) {
            sum += arr[i];
            if (i < k - 1) {
                continue;
            }
            if (sum >= k_thre) {
                ans++;
            }
            sum -= arr[i - k + 1];
        }
        return ans;
    }
};